[Bug]: CI tests fail due to embedded NATS/etcd shutdown and timeouts

## Describe the Bug

Continuous Integration (CI) tests are failing due to issues with embedded NATS in the project. Specifically, the test `TestNatsContext_Integration` fails with a timeout when waiting for a NATS message.

---

## Steps to Reproduce

1. Run the CI pipeline or execute `make test` locally.
2. Observe the failure of integration tests involving NATS.
3. Check the job logs for error details.

---

## Expected Behavior

All tests should pass successfully. Integration tests should be able to communicate with embedded NATS without timeouts or abrupt shutdown errors.

---

## Actual Behavior

- `TestNatsContext_Integration` fails with: `Timed out waiting for NATS message`


---

## Screenshots / Logs

```
2025/10/07 03:51:10 Starting embedded NATS server with JetStream...
2025/10/07 03:51:10 NATS server started on nats://0.0.0.0:35031
    nats_test.go:52: Timed out waiting for NATS message
--- FAIL: TestNatsContext_Integration (3.03s)
```

---

## Environment

- OS: [e.g., Ubuntu 22.04, Windows 11]
- CLI: [e.g., go 1.21, make]
- Project Version: [e.g., latest main, commit 111b5e980caa58b02329703228e3f56021918e84]
- Any custom configuration? [e.g., modified Docker compose file]

---

## Additional Context

This issue blocks contributors from passing CI tests, making it difficult to submit valid pull requests.

See job logs for more details: ref: [52110183220](https://github.com/open-ug/conveyor/actions/runs/18298612622/job/52110183220#step:5:322)

--- a/internal/handlers/resource_test.go
+++ b/internal/handlers/resource_test.go
@@ -3,9 +3,11 @@ package handlers_test
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"io"
 	"net/http"
 	"net/http/httptest"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -15,25 +17,57 @@ import (
 	"github.com/open-ug/conveyor/pkg/types"
 )
 
+func buildResourceDefinition(in int) types.ResourceDefinition {
+
+	return types.ResourceDefinition{
+		Name:        fmt.Sprintf("pipe:%d", in),
+		Description: "Pipeline resource definition",
+		Version:     "1.0.0",
+		Schema: map[string]any{
+			"properties": map[string]any{
+				"image": map[string]any{
+					"type": "string",
+				},
+				"steps": map[string]any{
+					"type": "array",
+					"items": map[string]any{
+						"type": "object",
+						"properties": map[string]any{
+							"name": map[string]any{
+								"type": "string",
+							},
+							"command": map[string]any{
+								"type": "string",
+							},
+						},
+						"required": []string{"name", "command"},
+					},
+				},
+			},
+			"required": []string{"steps"},
+		},
+	}
+}
+
 // Build example resource definition (based on your example)
 var resource_definition = types.ResourceDefinition{
 	Name:        "pipe4",
 	Description: "Pipeline resource definition",
 	Version:     "1.0.0",
-	Schema: map[string]interface{}{
-		"properties": map[string]interface{}{
-			"image": map[string]interface{}{
+	Schema: map[string]any{
+		"properties": map[string]any{
+			"image": map[string]any{
 				"type": "string",
 			},
-			"steps": map[string]interface{}{
+			"steps": map[string]any{
 				"type": "array",
-				"items": map[string]interface{}{
+				"items": map[string]any{
 					"type": "object",
-					"properties": map[string]interface{}{
-						"name": map[string]interface{}{
+					"properties": map[string]any{
+						"name": map[string]any{
 							"type": "string",
 						},
-						"command": map[string]interface{}{
+						"command": map[string]any{
 							"type": "string",
 						},
 					},
@@ -88,7 +122,8 @@ func Test_Resource_CRUD(t *testing.T) {
 
 	// --- Create Resource Definition ---
 	t.Run("create-resource-definition", func(t *testing.T) {
-		bodyBytes, _ := json.Marshal(resource_definition)
+		in := 2
+		bodyBytes, _ := json.Marshal(buildResourceDefinition(in))
 		req := httptest.NewRequest(http.MethodPost, "/resource-definitions", bytes.NewReader(bodyBytes))
 		req.Header.Set("Content-Type", "application/json")
 
@@ -97,12 +132,16 @@ func Test_Resource_CRUD(t *testing.T) {
 			t.Fatalf("create resource-definition request failed: %v", err)
 		}
 		defer resp.Body.Close()
-
 		respBody, _ := io.ReadAll(resp.Body)
+		if strings.Contains(string(respBody), "error") {
+			assert.Contains(t, string(respBody), "pipe:2 already exists")
+			return
+		}
 		assert.Equal(t, http.StatusCreated, resp.StatusCode, "expected 201 Created on create resource-definition")
 
 		var created types.ResourceDefinition
 		if assert.NoError(t, json.Unmarshal(respBody, &created), "unmarshal create resource-definition response") {
+			t.Logf("response body:%s\n", created)
 			assert.Equal(t, resource_definition.Name, created.Name)
 			assert.Equal(t, resource_definition.Version, created.Version)
 		}

--- a/internal/utils/etcd_test.go
+++ b/internal/utils/etcd_test.go
@@ -2,6 +2,7 @@ package utils_test
 
 import (
 	"context"
+	"os"
 	"testing"
 	"time"
 
@@ -10,7 +11,7 @@ import (
 )
 
 func TestNewEtcdClient_Integration(t *testing.T) {
-
+	os.Setenv("APP_ENV", "test")
 	client, err := utils.NewEtcdClient()
 	assert.NoError(t, err, "Expected to connect to etcd without error")
 	assert.NotNil(t, client)
@@ -44,4 +45,7 @@ func TestNewEtcdClient_Integration(t *testing.T) {
 	client.Cancel()
 	client.Client.Close()
 	client.ServerStop()
+	t.Cleanup(func() {
+		assert.NoError(t, os.Unsetenv("APP_ENV"), "error removing test app_env")
+	})
 }

--- a/internal/utils/nats.go
+++ b/internal/utils/nats.go
@@ -2,7 +2,9 @@ package utils
 
 import (
 	"context"
+	"fmt"
 	"log"
+	"os"
 	"time"
 
 	"github.com/nats-io/nats-server/v2/server"
@@ -20,6 +22,13 @@ type NatsContext struct {
 func NewNatsConn() *NatsContext {
 
 	conveyorDataDir := viper.GetString("api.data")
+	if conveyorDataDir == "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			fmt.Printf("error getting cwd:%v\n", err)
+		}
+		conveyorDataDir = cwd
+	}
 	dataDir := conveyorDataDir + "/nats"
 
 	var port int

--- a/internal/utils/nats_test.go
+++ b/internal/utils/nats_test.go
@@ -14,9 +14,8 @@ import (
 
 func TestNatsContext_Integration(t *testing.T) {
 	// Set test environment to enable random port allocation
-	originalEnv := os.Getenv("APP_ENV")
+	origEnv := os.Getenv("APP_ENV")
 	os.Setenv("APP_ENV", "test")
-	defer os.Setenv("APP_ENV", originalEnv) // Restore original value after test
 
 	config.InitConfig()
 
@@ -60,6 +59,10 @@ func TestNatsContext_Integration(t *testing.T) {
 	case <-time.After(3 * time.Second):
 		t.Fatal("Timed out waiting for NATS message")
 	}
+	t.Cleanup(func() {
+		assert.NoError(t, os.Unsetenv("APP_ENV"), "error unsetting app_env")
+		assert.NoError(t, os.Setenv("APP_ENV", origEnv), "error restoring app_env")
+	})
 }
 
 // helper to build consumer config

